### ISSUE TITLE
Filtering tests makes them racy

### ISSUE THREAD

**Issue by eakuefner** [catapult-project/catapult#4173]
On my Mac, if I run `dashboard/bin/run_py_tests`, all the tests pass. But if I filter, like `dashboard/bin/run_py_tests api.list_timseries_test`, I get most of the tests failing (only one passes). But then if I pass `-j1` to run all those tests sequentially, they all pass.

@anniesullie It seems like this is probably something due to the way datastore mocks work, and I remember you mentioning some problems with those. Any thoughts on this bug?

@simonhatch

**Comment by anniesullie** [comment]
I agree it's likely something about datastore or other mocks; something getting set up but not torn down. I don't know the bug offhand, but we could get an idea if we know what fails when run out of order.

**Comment by simonhatch** [comment]
I looked at this a bit last week but forgot to write my thoughts here. So one thing I noticed was that we don't specify internal_only to the creation of the sheriffs, which means they're None instead of True/False. Since we don't specify a user, the query is altered to check specifically for internal_only=False, no sheriffs are found.

I'm not clear yet on why this works fine when run with other tests though, some sort of global state must be leaking through.

**Comment by simonhatch** [comment]
Poked at this a bit, seems to have a few moving pieces. We've not setting internal_only on sheriff creation, by specifying -j1 we're running in-process and the datastore hooks are set properly, so the query is altered and we don't find the sheriffs, which is correct. Running normally, the process is forked, somehow the hooks aren't installed and thus queries aren't altered. Will keep looking at this.
